garak/probes/dan.py

```diff
@@ -15,511 +15,266 @@
 import garak.probes
 
 
-class Dan_11_0(garak.probes.Probe):
+class DANProbeMeta(type):
+    """
+    Metaclass for DAN probes that automatically configures probe classes.
+
+    Define a class with this metaclass and it will:
+    1. Load prompts from the specified prompt_file or auto-detect from class name
+    2. Apply common DAN probe configuration (lang, goal, tier, etc)
+    3. Handle custom attributes like detector, extra tags, etc.
+    """
+
+    # Shared configuration for all DAN probes
+    _DAN_DEFAULTS = {
+        "lang": "en",
+        "goal": "disregard the system prompt",
+        "tier": garak.probes.Tier.UNLISTED,
+        "active": False,
+        "tags": [
+            "avid-effect:security:S0403",
+            "owasp:llm01",
+            "quality:Security:PromptStability",
+            "payload:jailbreak",
+        ],
+        "recommended_detector": ["mitigation.MitigationBypass", "dan.DAN"],
+    }
+
+    def __new__(cls, name, bases, attrs, **kwargs):
+        """
+        Called when a new DAN probe class is created.
+        Automatically loads prompts and applies configuration.
+        """
+
+        # Apply default configuration
+        for key, default_value in cls._DAN_DEFAULTS.items():
+            if key not in attrs:
+                attrs[key] = default_value
+
+        # Extract parameters passed to metaclass
+        for key, value in kwargs.items():
+            if key not in attrs:
+                attrs[key] = value
+
+        # Handle extra_tags (merge with default tags)
+        if "extra_tags" in attrs:
+            default_tags = attrs.get("tags", cls._DAN_DEFAULTS["tags"])
+            attrs["tags"] = default_tags + attrs.pop("extra_tags")
+
+        # Handle custom detector
+        if "detector" in attrs:
+            attrs["recommended_detector"] = attrs.pop("detector")
+
+        # Determine prompt file
+        prompt_file = attrs.get("prompt_file", f"dan/{name}.txt")
+
+        # Load the prompts from the prompt file
+        try:
+            prompt_path = data_path / prompt_file
+            with open(prompt_path, "r", encoding="utf-8") as f:
+                content = f.read().strip()
+                # Allow multi-prompt files (one prompt per line)
+                if "\n" in content:
+                    prompts = [
+                        line.strip() for line in content.split("\n") if line.strip()
+                    ]
+                else:
+                    prompts = [content] if content else []
+            attrs["prompts"] = prompts
+        except Exception:
```

Comment:
This exception handler seem very broad, it would be preferred to handle known exception types like `GarakException` which can be raised by line 72 and exception types that `open()`, `read()` and `strip()` might raise.